aiscripts/mule.lib.evaluate_tradeoffers.xml

```diff
@@ -130,13 +136,24 @@
 						<continue />
 					</do_if>
 
-					<!-- how much profit for this trade? -->
-					<!-- If we use our own stations, we adjust the supply price by the sameFactionBuyMod to allow preferential treatment of player/owned supply chains -->
-					<do_if value="$selloffer.seller.owner == $shipEntity.owner">
-						<set_value name="$currentProfit" exact="$cargoHauled * ($buyoffer.unitprice - $selloffer.unitprice*$sameFactionBuyMod/100.0)" />
+					<do_if value="$dedicatedPlayerBuilder" >
+						<set_value name="$targetAmount" exact="$buyoffer.buyer.cargo.{$buyoffer.ware}.target" />
+						<!--calculate fill level region: 0%..33% = 3; 33%..67% = 2; 67%..100% = 1 -->
+						<!--REMIND (don't optimize) desired amount / target amount equals to the stock level when the construction is fully funded only-->
+						<set_value name="$fillLevelRegion" exact="((($buyoffer.desiredamount)f / ([$targetAmount,1].max)f - 0.0000001) * 3.0 + 1.0)i" />
+						<set_value name="$currentProfit" exact="$fillLevelRegion * 1Cr" />
```

Comment:
So the wares that are most needed have a profit of 3Cr, and the least needed a profit of 1Cr? 

I don't think this is much different from just letting the game manage the prices based on storage levels, is it?

Reply:
Profit in this case is a priority. Most needed will be supplied first. Plus a randomizer to fill the storage in equal quantities. Auto price management doesn't work when you are actively drain all the universe by building huge stations.